src/main/java/ProjectOcean/Controller/ScheduleController.java

```diff
@@ -63,15 +69,26 @@ private void displayAllYearsInSchedule() {
         }
     }
 
+    public void setVisualFeedbackForCoursePlacement(ICourse course){
+        for (YearController yearController: yearControllers) {
+            yearController.setGreenBorderColorInSlots(course.getStudyPeriod());
+        }
+    }
+
     private void updateControllerAccordingToModel() {
         years = model.getYears();
+        for (YearController yearController: yearControllers) {
+            yearController.deleteFromObserver();
+        }
+
         yearControllers.clear();
         int yearIndex = 0;
 
         for (IYear y :
                 years) {
             yearIndex++;
-            yearControllers.add(new YearController(y, model, moveDraggedObjectToCursor, addIconToScreen, yearIndex ));
+            //System.out.println("YEAR");
```

Comment:
Delete comment

Reply:
Could be done. But I think it might be more troubleshome than it is worth at this stage.